MozillaFoundation/foundation.mozilla.org
# Made petition campaign_id a required field

# Description


Related PRs/issues: #10579


**This PR does the following:**
- Makes the petition snippet's `campaign_id` field required. If a user tries to create or edit a petition without filling out this field, they will recieve a validation error.
- Updates `petition-thank-you.jsx` to pass the newly required `campaign_id` as a prop. The campaign ID is then appended to the donate modal's "donate_url". (Ex: `?form=donate&c_id=<id>`). That way FRU can use this ID for reporting purposes.
- Updates `002-thank-you.spec.js` to account for newly appended URL parameter.


# Screenshots:

**URL when hovering over "Yes, Ill chip in" Button:**

<img width="951" alt="Screenshot 2023-07-18 at 5 34 04 PM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/4cf175e8-65ef-491b-9e22-1dee8a1412cf">

**GTM FRU Donate Modal opened event:**

![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/a7d01319-ce62-49fc-b79d-05dc516fff51)



# Steps to test

## Testing that campaign_id is required

1. Copy the prod db on your local machine
2. Visit http://localhost:8000/cms/snippets/wagtailpages/petition/edit/306/
3. Delete what is set for `campaign_id` and click "Publish"
4. You should receive a validation error stating that Campaign ID is required


## Testing that the campaign_id is being sent to FRU/Salesforce
1. Check this branch out on your local machine
5. run `inv copy-prod-db`
6. Log into [google tag manager](https://tagmanager.google.com/#/container/accounts/6002697367/containers/36502666/workspaces/22) and click "preview"
7. In the URL box, enter http://localhost:8000/en/campaigns/tell-slack-make-dms-safe/?thank_you=true
8. On the new tab that appears, hover over the "Yes Ill chip in" button, note the `c_id` that is shown in the URL preview at the bottom left of the window.
9. Click the "Yes, I'll chip in" button to render the modal.
10. Go back to the GTM screen, and click the event " Event: FundraiseUp.checkoutOpen" on the left hand side of the screen, under "Summary".
11. Expand the details, the c_id should match the c_id that was shown on step five.


<img width="400" alt="Screenshot 2023-07-19 at 11 04 00 AM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/e62fc733-e72d-4387-a20c-9e1f28554f7a">


9. If everything is working as expected, testing is complete!
